src/core/screen.c

```diff
@@ -1237,16 +1238,16 @@ get_window_surface (MetaWindow *window)
   if (width > height)
     {
       max_size = MIN (MAX_PREVIEW_SIZE, current->rect.width / max_columns * MAX_PREVIEW_SCREEN_FRACTION);
-      ratio = ((double) width) / max_size;
+      ratio = max_size / width;
       width = (int) max_size;
-      height = (int) (((double) height) / ratio);
+      height = (int) (height * ratio + 0.5);
     }
   else
     {
       max_size = MIN (MAX_PREVIEW_SIZE, current->rect.height / max_columns * MAX_PREVIEW_SCREEN_FRACTION);
-      ratio = ((double) height) / max_size;
+      ratio = max_size / height;
```

Comment:
Why?

Reply:
performance and accuracy

Reply:
A division is more expensive than a product when working with floating point numbers, without counting the latency the order usually is 2:1 on single precision (float), a little more if it's on double precision (double).

The single precision numbers (float) have 7 decimal digits` log10(2^24) ≈ 7.225 `, you can find more information about float accuracy at https://en.wikipedia.org/wiki/Single-precision_floating-point_format

Reply:
Why are you adding unrelated changes in commit that fix build warning?

Reply:
IMHO, I think they are related but I may be wrong.

Reply:
From MR title you are fixing `-Wfloat-conversion`, no? It might be okayish if you would have included needed description in commit message.

And if you speak about performance include numbers. You gain nothing here, it is not performance critical path...

Maybe check - https://softwareengineering.stackexchange.com/questions/80084/is-premature-optimization-really-the-root-of-all-evil.

Reply:
I don't know, but I see it more feasible to use the factor than its inverse.